DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,153 @@
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<ResourceReferenceWithReferenceDetails>(response);
+            return result;
+        }
+
+        public async Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        )
+        {
+            const string bulkEndpoint = "/Resource/Bulk";
+            var endpointPath = learningHubApiBaseUrl + bulkEndpoint;
+            var requestUrl = GetUrlWithResourceReferenceIdQueryParams(endpointPath, resourceReferenceIds);
+
+            var response = await client.GetStringAsync(requestUrl);
+            var result = JsonConvert.DeserializeObject<BulkResourceReferences>(response);
+            return result;
+        }
+
+        private static string GetUrlWithSearchQueryParams(
+            string endpointPath,
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        )
+        {
+            var textQueryString = GetQueryString("text", text);
+            var offSetQueryString = GetQueryString("offset", offset.ToString());
+            var limitQueryString = GetQueryString("limit", limit.ToString());
+            var resourceTypesQueryString = GetMultipleQueryStrings("resourceTypes", resourceTypes);
+
+            var requestUrl =
+                $"{endpointPath}?{textQueryString}{offSetQueryString}{limitQueryString}{resourceTypesQueryString}";
+
+            requestUrl = GetUrlWithoutLastCharacter(requestUrl);
+
+            return requestUrl;
+        }
+
+        private static string GetUrlWithResourceReferenceIdQueryParams(
+            string endpointPath,
+            IEnumerable<int>? resourceIds = null
+        )
+        {
+            if (resourceIds == null)
+            {
+                return endpointPath;
+            }
+
+            var requestUrl = endpointPath + "?";
+
+            foreach (var resourceId in resourceIds)
+            {
+                requestUrl = requestUrl + "resourceReferenceIds=" + resourceId + "&";
+            }
+
+            requestUrl = GetUrlWithoutLastCharacter(requestUrl);
+
+            return requestUrl;
+        }
+
+        private static string GetMultipleQueryStrings(string key, IEnumerable<string>? values)
+        {
+            var queryString = "";
+
+            if (values == null)
+            {
+                return queryString;
+            }
+
+            foreach (var value in values)
+            {
+                queryString += string.IsNullOrEmpty(value) ? "" : $"{key}={value}&";
+            }
+
+            return queryString;
+        }
+
+        private static string GetQueryString(string key, string? value)
+        {
+            return string.IsNullOrEmpty(value) ? "" : $"{key}={value}&";
+        }
+
+        private static string GetUrlWithoutLastCharacter(string url)

```

Comment:
Particularly without needing to use this method...

Reply:
refactored so we don't use this anymore